chain/consensus/actors/atomic-replace/replace.go

```diff
@@ -0,0 +1,264 @@
+package replace
+
+// Sample actor that replaces the cid from one address to the other.
+// This actor is used as an example of how to use the actor execution
+// protocol.
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/cbor"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	actor "github.com/filecoin-project/lotus/chain/consensus/actors"
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical/atomic"
+	"github.com/filecoin-project/specs-actors/v7/actors/builtin"
+	"github.com/filecoin-project/specs-actors/v7/actors/runtime"
+	cid "github.com/ipfs/go-cid"
+	xerrors "golang.org/x/xerrors"
+)
+
+//go:generate go run ./gen/gen.go
+
+// example "Replace" actor that atomically replaces the cid from one owner
+// to the other.
+
+var _ runtime.VMActor = ReplaceActor{}
+var _ atomic.LockableActor = ReplaceActor{}
+
+// ReplaceState determines the actor state.
+// FIXME: We are using a non-efficient locking strategy for now
+// where the whole map is locked for an atomic execution.
+// We could use a more fine-grained approach, but this actor
+// has illustrative purposes for now. Consider improving it in
+// future iterations.
+type ReplaceState struct {
+	Owners *atomic.LockedState
+}
+
+type Owners struct {
+	M map[string]cid.Cid
```

Comment:
I believe that `M` is not a good name for public data types. It would work if `M` was used in a seminal paper, etc.
For example, `_, ok := o.M[k]` from below looks strange